sdk/appconfiguration/azure-appconfiguration-provider/azure/appconfiguration/provider/_azureappconfigurationprovider.py

```diff
@@ -52,8 +53,8 @@
 if TYPE_CHECKING:
     from azure.core.credentials import TokenCredential
 
-JSON = Union[str, Mapping[str, Any]]  # pylint: disable=unsubscriptable-object
-
+JSON = Mapping[str, Any]  # pylint: disable=unsubscriptable-object
```

Comment:
Based on how JSON is used, I don't think it was intended to be a Union here

Reply:
This was done based on how the Azure App Configuration service works, where everything it returns is a String as the value, but if it has the content type of JSON we make it JSON. @annatisch you helped me set this up do you remember why we did it this way?

Reply:
btw I'm not saying it shouldn't be Union[str, Mapping[str, Any]], it's just that where it's used, it would be redundant since we already say it's a union of str and mapping:

https://github.com/mrm9084/azure-sdk-for-python/blob/27426381b6274f5a39f240cc98b45cf14ff4aed0/sdk/appconfiguration/azure-appconfiguration-provider/azure/appconfiguration/provider/_azureappconfigurationprovider.py#L421

Reply:
Never noticed that. I'm good with changing it.